домашка спринт 12

Satturas opened the pull request.

vedmapishet committed:
удаление неиспользуемого перед ревью

vedmapishet committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	app/src/main/java/com/example/playlist_maker_dev/SearchActivity.kt

vedmapishet committed:
Merge remote-tracking branch 'origin/dev' into dev

vedmapishet committed:
реализовано переключение day/night на свитчере через sharedPreferences

vedmapishet committed:
добавлена верстка истории поиска

vedmapishet committed:
adding search history in progress...

vedmapishet committed:
adding search history in progress..... пока ругается на неинициализированный lateinit  и попытку length of null array

vedmapishet committed:
добавлено сохранение истории поиска

IliaAnK commented on app/src/main/res/layout/activity_search.xml:
```diff
@@ -49,7 +49,7 @@
             android:src="@drawable/search_image_vector" />
 
         <EditText
```
Рекомендация:
Иконку лупы можно объявить внутри `EditText` с помощью атрибутов:
```
android:drawableStart="@drawable/ic_search"
android:drawableTint="@color/someColor"
android:drawablePadding="8dp"
```

Но, сожалению, у `EditText` нет удобного API для изменения размеров иконки, поэтому нужно изначально добавить иконку правильного размера. + если иконку задать правильный цвет, то не придётся применять `android:drawableTint="@color/someColor"`

Satturas replied:
спасибо, буду разбираться

vedmapishet committed:
правки после ревью:
1. Устранен баг непоказа истории поиска при удалении поискового запроса с помощью клавиатуры
2. Исправлено отображение системной темы приложения при первом запуске на устройстве

Satturas commented:
Спасибо, баги с поиском и темой поправил. Рекомендации по верстке буду изучать и переделывать в дальнейшем.
